packages/data-model/src/types.ts

```diff
@@ -44,68 +115,106 @@ export type Institution = {
 	building?: string;
 	suite?: string;
 	city: string;
-	province: string;
+	province: string; // TODO: Handle Province or State?
 	postalCode: string;
 };
 
 export type Project = {
 	title: string;
 	website: string;
-	background: string;
+	abstract: string;
 	methodology: string;
 	summary: string;
-	relevantPublications: string;
+	publicationUrls: string[];
 };
 
-export type Revisions = {
+export type RevisionRequest = {
+	id: number; // TODO: Implement BigInt
+	applicationId: number; // TODO: Implement BigInt;
 	createdAt: Date;
-	createdBy: 'string';
+	createdBy: string;
 	version: number;
-	changes: {}[];
+	changes: {}[]; // TODO: Define structure, maybe ApplicationActionData[] ?
+	comments?: string;
+	applicantApproved: Boolean;
+	applicantNotes?: string;
+	institutionRepApproved: boolean;
+	institutionRepNotes?: string;
+	collaboratorsApproved: boolean;
+	collaboratorsNotes?: string;
+	projectApproved: boolean;
+	projectNotes?: string;
+	requestedStudiesApproved: boolean;
+	requestedStudiesNotes?: string;
 };
 
-export enum Status {
-	'DRAFT',
-	'REVIEW',
-	'APPROVED',
-	'CLOSED',
-	'REVOKED',
-}
-
-export type Application = {
-	status: keyof typeof Status;
-	applicant: Applicant;
-	institution: Institution;
-
-	institutional_representative: {
-		applicant: Applicant;
-		institution: Institution;
-	};
+export type ApplicationActionData = {
+	id: number; // TODO: Implement BigInt;
+	applicationId: number; // TODO: Implement BigInt;
+	createdAt: Date;
+	userId: string;
+	action: ApplicationActions;
+	stateBefore: ApplicationStates;
+	stateAfter: ApplicationStates;
+	revisionsRequestId: number; // TODO: Implement BigInt
+	// TODO: may need reference to a content diff
```

Comment:
Related to: https://github.com/Pan-Canadian-Genome-Library/daco/blob/main/docs/model/data-model.dbml#L143